pkg/resources/validation/namespace/namespace.go

```diff
@@ -0,0 +1,122 @@
+// Package namespace holds the Admit handler for webhook validation of requests modifying namespace objects
+package namespace
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/rancher/webhook/pkg/admission"
+	objectsv1 "github.com/rancher/webhook/pkg/generated/objects/core/v1"
+	"github.com/rancher/webhook/pkg/resources/validation"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	v1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	authorizationv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+	"k8s.io/utils/trace"
+)
+
+var gvr = schema.GroupVersionResource{
+	Version:  "v1",
+	Resource: "namespaces",
+}
+
+// Validator validates the namespace admission request.
+type Validator struct {
+	sar authorizationv1.SubjectAccessReviewInterface
+}
+
+// NewValidator returns a new validator used for validation of namespace requests.
+func NewValidator(sar authorizationv1.SubjectAccessReviewInterface) *Validator {
+	return &Validator{
+		sar: sar,
+	}
+}
+
+// GVR returns the GroupVersionKind for this CRD.
+func (v *Validator) GVR() schema.GroupVersionResource {
+	return gvr
```

Comment:
If you are not using the package-level variable `gvr` elsewhere, you can return it from the method directly.